### PR TITLE
Tom1 db connection

### PR DESCRIPTION
**Pull request by TomHo521**
Added many routes.  Please see documentation of my changes posted to the slack channel.

### CODE REVIEW

File: db/controllers.js [hunk 1]
```diff
@@ -1,83 +1,186 @@
...
       return err;
+    });
+};
+
+// adds a guest to a given session UNTESTED
+const updateRestaurant = function (obj_param) {
+  let {restaurant_id_api, restaurant_name, session_id} = obj_param;
+
+  let queryStr = `UPDATE "BOC_Sessions" SET restaurant_id_api = '${restaurant_id_api}', restaurant_name = '${restaurant_name}'
+                  WHERE (session_id = '${session_id}');`;
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply db/controllers.js]: Corrected in subsequent push up

File: db/controllers.js [hunk 2]
```diff
@@ -1,83 +1,186 @@
...
+
+  console.log(queryStr);
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply db/controllers.js]: Corrected in subsequent push up

File: db/controllers.js [hunk 3]
```diff
@@ -1,83 +1,186 @@
...
+
+  return pool.query(queryStr)
+    .then(res => {
+      //return JSON.stringify(res.rows, null, 2);
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply db/controllers.js]: Corrected in subsequent push up

File: db/controllers.js [hunk 4]
```diff
@@ -1,83 +1,186 @@
...
+    .catch(err => {
+      return err;
+    });
+};
+
+// creates a new user
+const createNewUser = function (obj_param) {
+  let { first_name, last_name, email, password } = obj_param;
+
+  let queryStr = `INSERT INTO "BOC_Users"(first_name, last_name, email, password) VALUES ('${first_name}', '${last_name}', '${email}', '${password}') RETURNING "user_id";`;
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply db/controllers.js]: Corrected in subsequent push up

File: server/user_db_routes/user_db_routes.js [hunk 5]
```diff
@@ -19,87 +19,275 @@ const testSession = require('./getSession');
...
-  })
-  .catch(err => {
-    res.status(400).send(err);
-  })
+  // let obj_params = {
+  //   session_name: 'Session Name #4',
+  //   restaurant_name: 'Restaurant #4', 
+  //   restaurant_id_api: '234234324324234',
+  //   host_id: 9
+  // };
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply server/user_db_routes/user_db_routes.js]: Corrected in subsequent push up

File: server/user_db_routes/user_db_routes.js [hunk 6]
```diff
@@ -19,87 +19,275 @@ const testSession = require('./getSession');
...
+      res.status(200).send(result);
+    })
+    .catch(err => {
+      res.status(400).send(err);
+    });
+});
+
+// adds an order with respect to a guest to a particular session
+router.post('/addOrder', (req, res) => {
+  // extract the proper parameters here
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply server/user_db_routes/user_db_routes.js:240]: Corrected in subsequent push up

File: server/user_db_routes/user_db_routes.js [hunk 7]
```diff
@@ -19,87 +19,275 @@ const testSession = require('./getSession');
...
+      res.status(200).send(result);
+    })
+    .catch(err => {
+      res.status(400).send(err);
+    });
+});
+
+// updates an order with respect to a guest to a particular session
+router.put('/updateOrder', (req, res) => {
+  // extract the proper parameters here
```
[the remark this answers is not in the document]
Comment (TomHo521) [reply server/user_db_routes/user_db_routes.js:264]: Corrected in subsequent push up

File: BOC_schema.sql [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+-- Table 'BOC_Sessions'
+-- 
+-- ---
+
+DROP TABLE IF EXISTS "BOC_Sessions" CASCADE;
+		
+CREATE TABLE "BOC_Sessions" (
+  session_id SERIAL,
+  host_id INTEGER,
+  rollover_session_id INTEGER,
```
Comment (MingHacker) [inline BOC_schema.sql:49]: what is difference between session_id and rollover_session_id?

File: BOC_schema.sql [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+-- ---
+-- Table 'BOC_Sessions'
+-- 
+-- ---
+
+DROP TABLE IF EXISTS "BOC_Sessions" CASCADE;
+		
+CREATE TABLE "BOC_Sessions" (
+  session_id SERIAL,
+  host_id INTEGER,
```
Comment (MingHacker) [inline BOC_schema.sql:48]: Probably consider adding foreign keys to make the tables related. 
sth like. foreign key (host_id) references BOC_Users(user_id)....

**Review by MingHacker — APPROVED** [review]

**Review by milofultz — APPROVED** [review]
There's stuff, but nothing that will break anything so lets get it out ASAP

File: BOC_schema.sql (hunk 9 above)
Comment (milofultz) [reply BOC_schema.sql:48]: agree. Lets get that in the backlog for the polish week

File: BOC_schema.sql (hunk 8 above)
Comment (milofultz) [reply BOC_schema.sql:49]: I think rollover_session_id is a stretch goal, so nothing important atm

**Pull request merged**